Make result.fail paramaters consistent with result.log (#943)
morganstanley/testplan: doc/newsfragments/2481_changed_result.fail_parameters.rst

> Make :py:meth:`fail <testplan.testing.multitest.result.Result.fail>` paramaters consistent with :py:meth:`log <testplan.testing.multitest.result.Result.log>`

Reviewer: Yeah, it's a breaking change. So we have to add more details in the newsfragment

> Make :py:meth:`fail <testplan.testing.multitest.result.Result.fail>` paramaters consistent with :py:meth:`log <testplan.testing.multitest.result.Result.log>`

Reviewer 2: ```suggestion
Make :py:meth:`fail <testplan.testing.multitest.result.Result.fail>` paramaters consistent with :py:meth:`log <testplan.testing.multitest.result.Result.log>. This is a breaking change if assertion was called previously as `result.fail(description=...)`. Simply ensure message is passed as it became a positional argument.
```

> Make :py:meth:`fail <testplan.testing.multitest.result.Result.fail>` paramaters consistent with :py:meth:`log <testplan.testing.multitest.result.Result.log>`. **breaking change** if assertion was called previously as `result.fail(description=...)`. Simply ensure message is passed as it became a positional argument.

Reviewer 3: Make :py:meth:`result.fail <testplan.testing.multitest.result.Result.fail>` parameters consistent with :py:meth:`result.log <testplan.testing.multitest.result.Result.log>`, that ``message`` is the required argument and ``description`` is optional.

> Make :py:meth:`fail <testplan.testing.multitest.result.Result.fail>` paramaters consistent with :py:meth:`log <testplan.testing.multitest.result.Result.log>`. **breaking change** if assertion was called previously as `result.fail(description=...)`. Simply ensure message is passed as it became a positional argument.

Reviewer 3: ```suggestion
Make :py:meth:`result.fail <testplan.testing.multitest.result.Result.fail>` parameters consistent with :py:meth:`result.log <testplan.testing.multitest.result.Result.log>`, that ``message`` is the required argument and ``description`` is optional.
```
